specification/communication/resource-manager/Microsoft.Communication/preview/2023-04-01-preview/CommunicationServices.json

```diff
@@ -686,7 +686,12 @@
           "x-ms-client-flatten": true
         },
         "identity": {
-          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity"
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/ManagedServiceIdentity",
+          "x-ms-mutability": [
+            "create",
+            "read",
+            "update"
+          ]
```

Comment:
I don't see why you need this, as by default that's the 3 values...
https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-mutability

But anyway I guess it won't hurt.

Reply:
That's when readonly property is indicated, also java sdk doesn't create builder methods to allow create or update when that isn't specified.

Reply:
But I didn't see anywhere in swagger stating that "identity" property is "readonly". The property is not in parrent "TrackedResource". And here it does not have `"readOnly": true`

Reply:
There's a third option when it's just null considering it's a JSON object, the fact that readonly isn't specified doesn't mean false is applied by default. Either case , this gave us our expected behavior using autorest for generating java package.

Reply:
Emm, I am not sure about it. I actually works on Java codegen, and a quick run on the spec (without this change) works fine.

```java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
// Code generated by Microsoft (R) AutoRest Code Generator.

package com.azure.resourcemanager.communication.generated;

import com.azure.resourcemanager.communication.models.ManagedServiceIdentity;
import com.azure.resourcemanager.communication.models.ManagedServiceIdentityType;

/** Samples for CommunicationServices CreateOrUpdate. */
public final class CommunicationServicesCreateOrUpdateSamples {
    /*
     * x-ms-original-file: specification/communication/resource-manager/Microsoft.Communication/preview/2023-04-01-preview/examples/communicationServices/createOrUpdate.json
     */
    /**
     * Sample code: Create or update resource.
     *
     * @param manager Entry point to CommunicationManager.
     */
    public static void createOrUpdateResource(com.azure.resourcemanager.communication.CommunicationManager manager) {
        manager
            .communicationServices()
            .define("MyCommunicationResource")
            .withRegion("Global")
            .withExistingResourceGroup("MyResourceGroup")
            .withDataLocation("United States")
            .create();
    }

    /*
     * x-ms-original-file: specification/communication/resource-manager/Microsoft.Communication/preview/2023-04-01-preview/examples/communicationServices/createOrUpdateWithSystemAssignedIdentity.json
     */
    /**
     * Sample code: Create or update resource with managed identity.
     *
     * @param manager Entry point to CommunicationManager.
     */
    public static void createOrUpdateResourceWithManagedIdentity(
        com.azure.resourcemanager.communication.CommunicationManager manager) {
        manager
            .communicationServices()
            .define("MyCommunicationResource")
            .withRegion("Global")
            .withExistingResourceGroup("MyResourceGroup")
            .withIdentity(new ManagedServiceIdentity().withType(ManagedServiceIdentityType.SYSTEM_ASSIGNED))
            .withDataLocation("United States")
            .create();
    }
}
```

Reply:
The update samples seems not correct, but code should still work if one do `withIdentity`.

Reply:
Interesting, this is worth investigating as 2 other people also don't have the withIdentity method generated.
What's your autorest version?

Reply:
`@autorest/java@4.1.17`

Use `--use=@autorest/java@4.1.17` to use this version.

Reply:
The sample is caused by the error in your example JSON.

The update JSON is
https://github.com/joannajiang112628/azure-rest-api-specs/blob/c9b373e139e99eb283da5a46196257a57b31e9fe/specification/communication/resource-manager/Microsoft.Communication/preview/2023-04-01-preview/examples/communicationServices/updateWithSystemAndUserIdentity.json#L12-L17

The `identity` property should be under `parameters`, like what you did in create example:
https://github.com/joannajiang112628/azure-rest-api-specs/blob/c9b373e139e99eb283da5a46196257a57b31e9fe/specification/communication/resource-manager/Microsoft.Communication/preview/2023-04-01-preview/examples/communicationServices/createOrUpdateWithSystemAssignedIdentity.json#L7-L15

Reply:
Hi Weidong, I rerun the Java SDK generation via the shared link https://github.com/Azure/azure-sdk-for-java/wiki/%5BManagement%5D--OpenAPI-specs for automated codegen. With or without the x-ms-mutability on the Json file, the withIdentity is actually shown as expected. I will rerun it again via autorest method and see if it gives the same outcome.